image/descriptor.go

```diff
@@ -66,22 +79,41 @@ func findDescriptor(w walker, name string) (*v1.Descriptor, error) {
 			return err
 		}
 
-		for i := 0; i < len(index.Manifests); i++ {
-			if index.Manifests[i].Annotations["org.opencontainers.ref.name"] == name {
-				d = index.Manifests[i]
-				return errEOW
+		switch argsParts[0] {
+		case "org.opencontainers.ref.name":
+			for i := 0; i < len(index.Manifests); i++ {
+				if index.Manifests[i].Annotations["org.opencontainers.ref.name"] == argsParts[1] {
+					refs = append(refs, index.Manifests[i])
+				}
+			}
+		case "platform.os":
+			for i := 0; i < len(index.Manifests); i++ {
+				if index.Manifests[i].Platform != nil && index.Manifests[i].Platform.OS == argsParts[1] {
+					refs = append(refs, index.Manifests[i])
+				}
 			}
+		case "digest":
```

Comment:
Walking the index to find an entry pointing at a given digest seems odd.  Doing that in `findDescriptor` seems fine, but `create` and `unpack` seem like they could skip straight over `index.json` into CAS resolution when you know the digest and media type (although you'd need to use `--type` or a new option to get the media type).

Reply:
What do you think about this [change](https://github.com/opencontainers/image-tools/pull/169/files#diff-da07f373046a0a24e67b18699dc15a70R177).

Reply:
> What do you think about this [change][1].

My problem wasn't with `validateDescriptor` for `digest==` ref names.  If we [get a media type from `--type`][2] and have a `digest==…` name, I think we can skip around [all of this][3] and never hit anything outside of CAS in `unpack`.

[1]: https://github.com/opencontainers/image-tools/pull/169/files#diff-da07f373046a0a24e67b18699dc15a70R177
[2]: https://github.com/opencontainers/image-tools/pull/169/files#r131313440
[3]: https://github.com/opencontainers/image-tools/blob/f3b94b31c161d74ce5012eb0a7cb4b9a6b95b3aa/image/image.go#L172-L179

Reply:
The type obtained by [--type](https://github.com/opencontainers/image-tools/blame/master/man/oci-image-tool-create.1.md#L27) is only the file type of `create` or `unpack` ( `imageLayout` or `image`). And the type of digest obtained here is manifest or index. We can't get the media type corresponding to digest through  `--type`, only through `findDescriptor` to confirm the media type.